From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AOS <aos@andyl.dev>
Date: Tue, 25 Feb 2026 00:00:00 +0000
Subject: [PATCH] aos: use system Python

Force Envoy to use system Python (AOS python3) instead of downloading
a prebuilt Python binary via rules_python toolchains.

Adapted from nixpkgs 0001-nixpkgs-use-system-Python.patch for Envoy 1.37.0.
---
 bazel/python_dependencies.bzl |  9 ++++-----
 bazel/repositories_extra.bzl  | 19 ++-----------------
 2 files changed, 6 insertions(+), 22 deletions(-)

diff --git a/bazel/python_dependencies.bzl b/bazel/python_dependencies.bzl
--- a/bazel/python_dependencies.bzl
+++ b/bazel/python_dependencies.bzl
@@ -3,25 +3,24 @@ load("@envoy_toolshed//:packages.bzl", "load_packages")
 load("@rules_python//python:pip.bzl", "pip_parse")

 def envoy_python_dependencies():
-    # TODO(phlax): rename base_pip3 -> pip3 and remove this
-    load_packages()
+    pip_parse(
+        name = "pip3",
+        requirements_lock = "@envoy_toolshed//:requirements.txt",
+    )
     pip_parse(
         name = "base_pip3",
-        python_interpreter_target = "@python3_12_host//:python",
         requirements_lock = "@envoy//tools/base:requirements.txt",
         extra_pip_args = ["--require-hashes"],
     )

     pip_parse(
         name = "dev_pip3",
-        python_interpreter_target = "@python3_12_host//:python",
         requirements_lock = "@envoy//tools/dev:requirements.txt",
         extra_pip_args = ["--require-hashes"],
     )

     pip_parse(
         name = "fuzzing_pip3",
-        python_interpreter_target = "@python3_12_host//:python",
         requirements_lock = "@rules_fuzzing//fuzzing:requirements.txt",
         extra_pip_args = ["--require-hashes"],
     )
diff --git a/bazel/repositories_extra.bzl b/bazel/repositories_extra.bzl
--- a/bazel/repositories_extra.bzl
+++ b/bazel/repositories_extra.bzl
@@ -5,21 +5,13 @@ load("@envoy_examples//bazel:env.bzl", "envoy_examples_env")
 load("@envoy_toolshed//sysroot:sysroot.bzl", "setup_sysroots")
 load("@proxy_wasm_cpp_host//bazel/cargo/wasmtime/remote:crates.bzl", "crate_repositories")
 load("@rules_cc//cc:extensions.bzl", "compatibility_proxy_repo")
-load("@rules_python//python:repositories.bzl", "py_repositories", "python_register_toolchains")
+load("@rules_python//python:repositories.bzl", "py_repositories")
 load("@toolchains_llvm//toolchain:deps.bzl", "bazel_toolchain_dependencies")
 load("//bazel/external/cargo:crates.bzl", "raze_fetch_remote_crates")

-def _python_minor_version(python_version):
-    return "_".join(python_version.split(".")[:-1])
-
-GLIBC_VERSION = "2.31"
-
-# Python version for `rules_python`
-PYTHON_VERSION = "3.12.3"
-PYTHON_MINOR_VERSION = _python_minor_version(PYTHON_VERSION)
+GLIBC_VERSION = "2.31"

 # Envoy deps that rely on a first stage of dependency loading in envoy_dependencies().
 def envoy_dependencies_extra(
         glibc_version = GLIBC_VERSION,
-        python_version = PYTHON_VERSION,
         ignore_root_user_error = False):
     compatibility_proxy_repo()
     bazel_toolchain_dependencies()
@@ -29,13 +21,6 @@ def envoy_dependencies_extra(
     crate_repositories()
     py_repositories()

-    # Registers underscored Python minor version - eg `python3_10`
-    python_register_toolchains(
-        name = "python%s" % _python_minor_version(python_version),
-        python_version = python_version,
-        ignore_root_user_error = ignore_root_user_error,
-    )
-
     aspect_bazel_lib_dependencies()

     if not native.existing_rule("proto_bazel_features"):
